tfimm/train/train.py

```diff
@@ -25,8 +25,8 @@ class ExperimentConfig:
     # Data
     train_dataset: Any
     train_dataset_class: str
-    val_dataset: Any
-    val_dataset_class: str
+    val_dataset: Optional[Any] = None
+    val_dataset_class: Optional[str] = None
```

Comment:
The difficulty with `Optional` and config classes is that I am using the typehint as type information for parsing command line arguments. Which is why types for config classes are fairly strict. We can still have the val dataset optional by passing the empty string as `val_dataset_class`.

And for `Any` you don't need `Optional`. `Any` covers everything already.

Reply:
Thanks for the comments. We still need to provide default values, so I now have
`val_dataset: Any = None` and `val_dataset_class: str = ""`

Reply:
Yes, that makes sense. We also need to adjust config parsing to allow for empty strings for `_class` fields. I will do that now in the `feat/config` branch.

Thanks for the PR.